src/Raven.Server/Documents/Indexes/IndexStore.cs

```diff
@@ -1004,14 +1003,84 @@ public IEnumerable<Index> GetIndexes()
 
         public void RunIdleOperations()
         {
-            AsyncHelpers.RunSync(HandleUnusedAutoIndexes);
+            DatabaseRecord record;
+            using (_serverStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+            using (context.OpenReadTransaction())
+                record = _serverStore.Cluster.ReadDatabase(context, _documentDatabase.Name);
+
+            if (record.Topology.Members[0] != _serverStore.NodeTag)
+                return;
+
+            AsyncHelpers.RunSync(RunIdleOperationsAsync);
         }
 
-        private async Task HandleUnusedAutoIndexes()
+        private async Task RunIdleOperationsAsync()
+        {
+            await HandleUnusedAutoIndexes();
+            await DeleteSurpassedAutoIndexes();
+        }
+
+        private async Task DeleteSurpassedAutoIndexes()
```

Comment:
Let's run this IFF there have been changes in the indexes since we last run this.

Reply:
Can just check the index of the db record to see if it changed

Reply:
Done